packages/devtools_app/lib/src/shared/config_specific/server/_server_web.dart

```diff
@@ -337,6 +338,31 @@ DevToolsJsonFile _devToolsJsonFileFromResponse(
   );
 }
 
+Future<List<DevToolsExtensionConfig>> refreshAvailableExtensions(
+  String? rootPath,
+) async {
+  if (isDevToolsServerAvailable) {
+    final uri = Uri(
+      path: apiServeAvailableExtensions,
+      queryParameters: {extensionRootPathPropertyName: rootPath},
+    );
+    final resp = await request(uri.toString());
+    if (resp?.status == HttpStatus.ok) {
+      final parsedResult = json.decode(resp!.responseText!);
+      final extensionsAsJson =
+          (parsedResult[extensionsResultPropertyName]! as List<Object?>)
+              .cast<Map<String, Object?>>();
```

Comment:
There's a possibility this cast could fail. If the result is `List<Object?>`, casting to a non-nullable `List<Map<String, Object?>>` would throw if there is a null entry. It probably doesn't make much of a difference since we'd throw during the `as List<Object>` cast if there's a null entry in the list, but can we instead do `(result as List<Object>).cast<Map<String, Object?>>()` to make it clear that we don't expect a null entry?

Reply:
Replacing `List<Object?>` with `List<Object>` resulted in this error:
```
TypeError: Instance of 'JSArray<dynamic>': type 'JSArray<dynamic>' is not a subtype of type 'List<Object>'
```

Instead, I added a `.whereNotNull()` call in the chain to ensure we aren't casting a potentially null element to a non null map.

Reply:
Huh, that's interesting... Using `.whereNotNull()` is a good call.